document-generator-prosecution/src/test/java/uk/gov/companieshouse/document/generator/prosecution/service/ProsecutionServiceTest.java

```diff
@@ -0,0 +1,187 @@
+import uk.gov.companieshouse.api.handler.prosecution.defendant.PrivateDefendantResourceHandler;
+import uk.gov.companieshouse.api.handler.prosecution.defendant.request.PrivateDefendantGet;
+import uk.gov.companieshouse.api.handler.prosecution.offence.PrivateOffenceResourceHandler;
+import uk.gov.companieshouse.api.handler.prosecution.offence.request.PrivateOffenceList;
+import uk.gov.companieshouse.api.handler.prosecution.prosecutioncase.PrivateProsecutionCaseResourceHandler;
+import uk.gov.companieshouse.api.handler.prosecution.prosecutioncase.request.PrivateProsecutionCaseGet;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.api.model.prosecution.defendant.DefendantApi;
+import uk.gov.companieshouse.api.model.prosecution.offence.OffenceApi;
+import uk.gov.companieshouse.api.model.prosecution.prosecutioncase.ProsecutionCaseApi;
+import uk.gov.companieshouse.document.generator.prosecution.exception.ProsecutionServiceException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class ProsecutionServiceTest {
+
+    private static final String PROSECUTION_CASE_URI = "/internal/company/1234/prosecution-cases/4321";
+    private static final String DEFENDANT_URI = PROSECUTION_CASE_URI + "/defendants/1122";
+    private static final String OFFENCES_URI = DEFENDANT_URI + "/offences";
+
+    private DefendantApi defendantApi = new DefendantApi();
+    private ProsecutionCaseApi prosecutionCaseApi = new ProsecutionCaseApi();
+    private OffenceApi[] offenceApis = new OffenceApi[1];
+
+    @Mock
+    private InternalApiClient internalApiClient;
+
+    @Mock
+    private PrivateDefendantResourceHandler defendantHandler;
+
+    @Mock
+    private ApiResponse<DefendantApi> defendantResponse;
+
+    @Mock
+    private PrivateDefendantGet defendantGet;
+
+    @Mock
+    private PrivateProsecutionCaseResourceHandler prosecutionCaseHandler;
+
+    @Mock
+    private ApiResponse<ProsecutionCaseApi> prosecutionCaseApiResponse;
+
+    @Mock
+    private PrivateProsecutionCaseGet prosecutionCaseGet;
+
+    @Mock
+    private PrivateOffenceResourceHandler offenceHandler;
+
+    @Mock
+    private ApiResponse<OffenceApi[]> offenceApiResponse;
+
+    @Mock
+    private PrivateOffenceList offenceList;
+
+    @Mock
+    private ApiClientService apiClientService;
+
+    @InjectMocks
+    private ProsecutionService prosecutionService;
+
+    @BeforeEach void setUp() {
+        when(apiClientService.getInternalApiClient()).thenReturn(internalApiClient);
+    }
+
+    @Test
+    @DisplayName("Tests successful get of DefendantApi")
+    void getDefendantApiSuccessful() throws URIValidationException, ApiErrorResponseException, ProsecutionServiceException {
```

Comment:
No need to make any changes, just an FYI. In my opinion the unit tests could thrown just an Exception to make them quicker to write and easier to maintain. We don't need to handle the different exception types from the unit tests. Saves someone updating every unit test if an exception changes :)